While setting MWM_HINTS_DECORATIONS to disable window decorations
seems to work with most window managers, using _MOTIF_WM_INFO seems
not to, and in some cases can lead to severe problems with focus
management. Therefore this patch here disables the _MOTIF_WM_INFO
check and always attempts to have the window manager honour the
decorations hint.

If your window manager does not honour MWM_HINTS_DECORATIONS, and you
are determined that you want to have borderless windows nevertheless,
you are still free to enable override-redirect manually, using the
corresponding command line option or X resource.

See also: http://bugs.gentoo.org/show_bug.cgi?id=237271
2008-09-10 Martin von Gagern

--- a/src/init.C	15 Jul 2008 16:41:22 -0000	1.253
+++ b/src/init.C	10 Sep 2008 08:07:09 -0000
@@ -1078,15 +1078,7 @@
 #if ENABLE_FRILLS
   if (option (Opt_borderLess))
     {
-      if (XInternAtom (dpy, "_MOTIF_WM_INFO", True) == None)
-        {
-          // rxvt_warn("Window Manager does not support MWM hints.  Bypassing window manager control for borderless window.\n");
-          attributes.override_redirect = true;
-        }
-      else
-        {
           mwmhints.flags = MWM_HINTS_DECORATIONS;
-        }
     }
 #endif
 
